client/packages/programs/src/JsonForms/components/BloodPressure.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect } from 'react';
+import {
+  ControlProps,
+  composePaths,
+  rankWith,
+  subErrorsAt,
+  uiTypeIs,
+} from '@jsonforms/core';
+import {
+  Box,
+  DetailInputWithLabelRow,
+  NumUtils,
+  NumericTextInput,
+  Typography,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { FORM_LABEL_WIDTH } from '../common';
+import { useJsonForms, withJsonFormsControlProps } from '@jsonforms/react';
+import { useJSONFormsCustomError } from '../common/hooks/useJSONFormsCustomError';
+
+export const bloodPressureTester = rankWith(10, uiTypeIs('BloodPressure'));
+
+export const UIComponent = (props: ControlProps) => {
+  const t = useTranslation('programs');
+  const { data, handleChange, label, path, schema } = props;
+  const { core } = useJsonForms();
+  const [systolic, setSystolic] = React.useState<number | undefined>(undefined);
+  const [diastolic, setDiastolic] = React.useState<number | undefined>(
+    undefined
+  );
+  const systolicPath = composePaths(path, 'systolic');
+  const diastolicPath = composePaths(path, 'diastolic');
+  const { customError, setCustomError } = useJSONFormsCustomError(
+    path,
+    'BloodPressure'
+  );
+
+  useEffect(() => {
+    if (core) {
+      const getChildErrors = subErrorsAt(path, schema);
+      const errors = getChildErrors(core);
+      setCustomError(errors[0]?.message);
+    }
+  }, [core]);
+
+  if (!props.visible) {
+    return null;
+  }
```

Comment:
You are not allowed to conditionally call a hook, i.e. you return here and then call useEffect later. You can just move this return down just before returning the "tsx html".

There should already be an active lint rule for it...